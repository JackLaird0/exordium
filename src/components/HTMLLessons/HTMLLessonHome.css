.key-features {
  font-weight: 100;
}

.html-image {
  height: 200px;
  margin: 20px;
}