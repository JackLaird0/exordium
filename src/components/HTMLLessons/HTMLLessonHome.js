import React, { Component } from 'react';

export class HTMLLessonHome extends Component {

  render() {
    return (
      <main>
        <header>
          <h1>Welcome to HTML!</h1>
        </header>
        <div className="basics">
          <h1>Before you start, please read this brief overview</h1>
          <p>HTML is the backbone of our application</p>
          <p>The key features of HTML include:</p>
          <li>Elements</li>
          <li>Tags</li>
          <li>Attributes</li>
          <li>Classes and Id's</li>
          <p>Understanding these four features will allow you to build applications with solid structure and content. Lay the foundations of your website here.</p>
          <p>Let's get started!</p>
        </div>
        <div className="instructions">
          <p>To get started, choose one of the lessons below</p>
          <p>After you have completed the lesson, play the game to test your knowledge!</p>
        </div>
        <div className="lessons">
          <div>Lesson 01: Elements & Tags</div>
          <div>Lesson 02: Classes & Ids</div>
          <div>Lesson 03: Using Attributes</div>
          <div>Lesson 04: Advanced HTML Tags</div>
        </div>
        <div className="games">
          <div>Game 01: Elements & Tags</div>
          <div>Game 02: Declaring Functions</div>
          <div>Game 03: Using Attributes</div>
          <div>Game 04: Advanced HTML Tags</div>
        </div>
      </main>
    )
  }
}

export default HTMLLessonHome;