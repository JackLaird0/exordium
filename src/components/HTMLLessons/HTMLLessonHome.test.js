import React from 'react';
import { shallow } from 'enzyme';
import HTMLLessonHome from './HTMLLessonHome';

describe('HTMLLessonHome', () => {

  it('should match the snapshot', () => {
    let wrapper = shallow(<HTMLLessonHome />)
    expect(wrapper).toMatchSnapshot();
  })
})