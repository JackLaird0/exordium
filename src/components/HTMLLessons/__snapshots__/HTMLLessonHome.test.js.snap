// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTMLLessonHome should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HTMLLessonHome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1>
                    Welcome to HTML!
          </h1>
</header>,
        <div
          className="basics"
>
          <h1>
                    Before you start, please read this brief overview
          </h1>
          <p>
                    HTML is the backbone of our application
          </p>
          <p>
                    The key features of HTML include:
          </p>
          <li>
                    Elements
          </li>
          <li>
                    Tags
          </li>
          <li>
                    Attributes
          </li>
          <li>
                    Classes and Id's
          </li>
          <p>
                    Understanding these four features will allow you to build applications with solid structure and content. Lay the foundations of your website here.
          </p>
          <p>
                    Let's get started!
          </p>
</div>,
        <div
          className="instructions"
>
          <p>
                    To get started, choose one of the lessons below
          </p>
          <p>
                    After you have completed the lesson, play the game to test your knowledge!
          </p>
</div>,
        <div
          className="lessons"
>
          <div>
                    Lesson 01: Elements & Tags
          </div>
          <div>
                    Lesson 02: Classes & Ids
          </div>
          <div>
                    Lesson 03: Using Attributes
          </div>
          <div>
                    Lesson 04: Advanced HTML Tags
          </div>
</div>,
        <div
          className="games"
>
          <div>
                    Game 01: Elements & Tags
          </div>
          <div>
                    Game 02: Declaring Functions
          </div>
          <div>
                    Game 03: Using Attributes
          </div>
          <div>
                    Game 04: Advanced HTML Tags
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Welcome to HTML!
</h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to HTML!",
          },
          "ref": null,
          "rendered": "Welcome to HTML!",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Before you start, please read this brief overview
</h1>,
            <p>
              HTML is the backbone of our application
</p>,
            <p>
              The key features of HTML include:
</p>,
            <li>
              Elements
</li>,
            <li>
              Tags
</li>,
            <li>
              Attributes
</li>,
            <li>
              Classes and Id's
</li>,
            <p>
              Understanding these four features will allow you to build applications with solid structure and content. Lay the foundations of your website here.
</p>,
            <p>
              Let's get started!
</p>,
          ],
          "className": "basics",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Before you start, please read this brief overview",
            },
            "ref": null,
            "rendered": "Before you start, please read this brief overview",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "HTML is the backbone of our application",
            },
            "ref": null,
            "rendered": "HTML is the backbone of our application",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The key features of HTML include:",
            },
            "ref": null,
            "rendered": "The key features of HTML include:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Elements",
            },
            "ref": null,
            "rendered": "Elements",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tags",
            },
            "ref": null,
            "rendered": "Tags",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Attributes",
            },
            "ref": null,
            "rendered": "Attributes",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Classes and Id's",
            },
            "ref": null,
            "rendered": "Classes and Id's",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Understanding these four features will allow you to build applications with solid structure and content. Lay the foundations of your website here.",
            },
            "ref": null,
            "rendered": "Understanding these four features will allow you to build applications with solid structure and content. Lay the foundations of your website here.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Let's get started!",
            },
            "ref": null,
            "rendered": "Let's get started!",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              To get started, choose one of the lessons below
</p>,
            <p>
              After you have completed the lesson, play the game to test your knowledge!
</p>,
          ],
          "className": "instructions",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "To get started, choose one of the lessons below",
            },
            "ref": null,
            "rendered": "To get started, choose one of the lessons below",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "After you have completed the lesson, play the game to test your knowledge!",
            },
            "ref": null,
            "rendered": "After you have completed the lesson, play the game to test your knowledge!",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Lesson 01: Elements & Tags
</div>,
            <div>
              Lesson 02: Classes & Ids
</div>,
            <div>
              Lesson 03: Using Attributes
</div>,
            <div>
              Lesson 04: Advanced HTML Tags
</div>,
          ],
          "className": "lessons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lesson 01: Elements & Tags",
            },
            "ref": null,
            "rendered": "Lesson 01: Elements & Tags",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lesson 02: Classes & Ids",
            },
            "ref": null,
            "rendered": "Lesson 02: Classes & Ids",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lesson 03: Using Attributes",
            },
            "ref": null,
            "rendered": "Lesson 03: Using Attributes",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lesson 04: Advanced HTML Tags",
            },
            "ref": null,
            "rendered": "Lesson 04: Advanced HTML Tags",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Game 01: Elements & Tags
</div>,
            <div>
              Game 02: Declaring Functions
</div>,
            <div>
              Game 03: Using Attributes
</div>,
            <div>
              Game 04: Advanced HTML Tags
</div>,
          ],
          "className": "games",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Game 01: Elements & Tags",
            },
            "ref": null,
            "rendered": "Game 01: Elements & Tags",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Game 02: Declaring Functions",
            },
            "ref": null,
            "rendered": "Game 02: Declaring Functions",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Game 03: Using Attributes",
            },
            "ref": null,
            "rendered": "Game 03: Using Attributes",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Game 04: Advanced HTML Tags",
            },
            "ref": null,
            "rendered": "Game 04: Advanced HTML Tags",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
                        Welcome to HTML!
            </h1>
</header>,
          <div
            className="basics"
>
            <h1>
                        Before you start, please read this brief overview
            </h1>
            <p>
                        HTML is the backbone of our application
            </p>
            <p>
                        The key features of HTML include:
            </p>
            <li>
                        Elements
            </li>
            <li>
                        Tags
            </li>
            <li>
                        Attributes
            </li>
            <li>
                        Classes and Id's
            </li>
            <p>
                        Understanding these four features will allow you to build applications with solid structure and content. Lay the foundations of your website here.
            </p>
            <p>
                        Let's get started!
            </p>
</div>,
          <div
            className="instructions"
>
            <p>
                        To get started, choose one of the lessons below
            </p>
            <p>
                        After you have completed the lesson, play the game to test your knowledge!
            </p>
</div>,
          <div
            className="lessons"
>
            <div>
                        Lesson 01: Elements & Tags
            </div>
            <div>
                        Lesson 02: Classes & Ids
            </div>
            <div>
                        Lesson 03: Using Attributes
            </div>
            <div>
                        Lesson 04: Advanced HTML Tags
            </div>
</div>,
          <div
            className="games"
>
            <div>
                        Game 01: Elements & Tags
            </div>
            <div>
                        Game 02: Declaring Functions
            </div>
            <div>
                        Game 03: Using Attributes
            </div>
            <div>
                        Game 04: Advanced HTML Tags
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Welcome to HTML!
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to HTML!",
            },
            "ref": null,
            "rendered": "Welcome to HTML!",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Before you start, please read this brief overview
</h1>,
              <p>
                HTML is the backbone of our application
</p>,
              <p>
                The key features of HTML include:
</p>,
              <li>
                Elements
</li>,
              <li>
                Tags
</li>,
              <li>
                Attributes
</li>,
              <li>
                Classes and Id's
</li>,
              <p>
                Understanding these four features will allow you to build applications with solid structure and content. Lay the foundations of your website here.
</p>,
              <p>
                Let's get started!
</p>,
            ],
            "className": "basics",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Before you start, please read this brief overview",
              },
              "ref": null,
              "rendered": "Before you start, please read this brief overview",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "HTML is the backbone of our application",
              },
              "ref": null,
              "rendered": "HTML is the backbone of our application",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "The key features of HTML include:",
              },
              "ref": null,
              "rendered": "The key features of HTML include:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Elements",
              },
              "ref": null,
              "rendered": "Elements",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tags",
              },
              "ref": null,
              "rendered": "Tags",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Attributes",
              },
              "ref": null,
              "rendered": "Attributes",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Classes and Id's",
              },
              "ref": null,
              "rendered": "Classes and Id's",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Understanding these four features will allow you to build applications with solid structure and content. Lay the foundations of your website here.",
              },
              "ref": null,
              "rendered": "Understanding these four features will allow you to build applications with solid structure and content. Lay the foundations of your website here.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Let's get started!",
              },
              "ref": null,
              "rendered": "Let's get started!",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                To get started, choose one of the lessons below
</p>,
              <p>
                After you have completed the lesson, play the game to test your knowledge!
</p>,
            ],
            "className": "instructions",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "To get started, choose one of the lessons below",
              },
              "ref": null,
              "rendered": "To get started, choose one of the lessons below",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "After you have completed the lesson, play the game to test your knowledge!",
              },
              "ref": null,
              "rendered": "After you have completed the lesson, play the game to test your knowledge!",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Lesson 01: Elements & Tags
</div>,
              <div>
                Lesson 02: Classes & Ids
</div>,
              <div>
                Lesson 03: Using Attributes
</div>,
              <div>
                Lesson 04: Advanced HTML Tags
</div>,
            ],
            "className": "lessons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lesson 01: Elements & Tags",
              },
              "ref": null,
              "rendered": "Lesson 01: Elements & Tags",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lesson 02: Classes & Ids",
              },
              "ref": null,
              "rendered": "Lesson 02: Classes & Ids",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lesson 03: Using Attributes",
              },
              "ref": null,
              "rendered": "Lesson 03: Using Attributes",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lesson 04: Advanced HTML Tags",
              },
              "ref": null,
              "rendered": "Lesson 04: Advanced HTML Tags",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Game 01: Elements & Tags
</div>,
              <div>
                Game 02: Declaring Functions
</div>,
              <div>
                Game 03: Using Attributes
</div>,
              <div>
                Game 04: Advanced HTML Tags
</div>,
            ],
            "className": "games",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Game 01: Elements & Tags",
              },
              "ref": null,
              "rendered": "Game 01: Elements & Tags",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Game 02: Declaring Functions",
              },
              "ref": null,
              "rendered": "Game 02: Declaring Functions",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Game 03: Using Attributes",
              },
              "ref": null,
              "rendered": "Game 03: Using Attributes",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Game 04: Advanced HTML Tags",
              },
              "ref": null,
              "rendered": "Game 04: Advanced HTML Tags",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
